components/src/components/molecules/Select/Select.tsx

```diff
@@ -89,42 +141,105 @@ const Chevron = styled(IconDownIndicatorSvg)<{
   `,
 )
 
-const SelectOptionContainer = styled.div<{ $open?: boolean }>(
-  ({ theme, $open }) => css`
-    display: flex;
-    flex-direction: column;
-    align-items: flex-start;
-    justify-content: space-between;
+const SelectOptionContainer = styled.div<{
+  $state?: TransitionState
+  $direction?: Direction
+  $rows?: number
+}>(
+  ({ theme, $state, $direction, $rows }) => css`
+    display: block;
     position: absolute;
     visibility: hidden;
     opacity: 0;
     overflow: hidden;
 
     margin-top: ${theme.space['1.5']};
     padding: ${theme.space['1.5']};
-    width: ${theme.space['full']};
+    min-width: ${theme.space['full']};
     height: ${theme.space['fit']};
     border-radius: ${theme.radii['medium']};
     box-shadow: ${theme.boxShadows['0.02']};
     background: ${theme.colors.background};
 
-    ${$open
+    ${$state === 'entered'
       ? css`
           z-index: 20;
           visibility: visible;
-          margin-top: ${theme.space['1.5']};
+          top: ${$direction === 'up'
+            ? `auto`
+            : `calc(100% + ${theme.space['1.5']})`};
+          bottom: ${$direction === 'up'
+            ? `calc(100% + ${theme.space['1.5']})`
+            : 'auto'};
           opacity: ${theme.opacity['100']};
-          transition: all 0.3s cubic-bezier(1, 0, 0.22, 1.6),
+          transition: all 0.3s ${theme.transitionTimingFunction.popIn},
```

Comment:
popIn transition =/= the old transition, unless this was intentional in which case the old one looks better imo.